vapoursynth/src/api.rs

```diff
@@ -675,10 +788,26 @@ impl API {
     /// # Safety
     /// The caller must ensure `core` is valid.
     #[inline]
+    #[cfg(not(feature = "gte-vapoursynth-api-36"))]
     pub(crate) unsafe fn get_core_info(self, core: *mut ffi::VSCore) -> *const ffi::VSCoreInfo {
         (self.handle.as_ref().getCoreInfo)(core)
     }
 
+    /// Returns information about the VapourSynth core.
+   ///
+   /// # Safety
+   /// The caller must ensure `core` is valid.
+    #[inline]
+    #[cfg(feature = "gte-vapoursynth-api-36")]
+    pub(crate) unsafe fn get_core_info(self, core: *mut ffi::VSCore) -> *const ffi::VSCoreInfo {
+        use std::mem::MaybeUninit;
+
+        let mut core_info = MaybeUninit::<ffi::VSCoreInfo>::uninit();
+        (self.handle.as_ref().getCoreInfo2)(core, core_info.as_mut_ptr());
+        core_info.assume_init();
```

Comment:
As far as I understand, `assume_init()` doesn't do anything on its own, the init assumption comes from simply the fact that a `T` is accessed in the union.